app/controllers/application_controller.rb

```diff
@@ -198,8 +214,7 @@ def redirect_back_or(path)
 
   def redirect_to_siginin_if_anon_or(path)
     if current_user.nil?
-      session[:redirect_path_after_signin] = request.path
-      redirect_to new_user_session_path
+      redirect_to auth_login_path(after_sign_in_path: request.path)
     elsif !path.empty?
```

Comment:
How about we remove this method and move it into pundit_user_not_authorized, the basic refactoring would avoid the possibility of a non redirect:
```
if current_user.nil?
  redirect_to auth_login_path(after_sign_in_path: request.path)
else
  redirect_to after_sign_in_path_for(current_user)
end
```